cli/crates/typed-resolvers/src/codegen.rs

```diff
@@ -0,0 +1,149 @@
+use crate::analyze::{AnalyzedSchema, BuiltinScalar, Definition, Field, FieldTypeKind, ListWrapper, ObjectKind};
+use std::fmt;
+
+const INDENT: &str = "  ";
+
+pub(crate) fn generate_module<O>(schema: &AnalyzedSchema<'_>, out: &mut O) -> fmt::Result
+where
+    O: fmt::Write,
+{
+    for (idx, definition) in schema.definitions.iter().enumerate() {
+        match definition {
+            Definition::Object(id) => {
+                let object_type = &schema[*id];
+                let graphql_type_name = object_type.name;
+                let ts_type_name = safe_ts_type_name(graphql_type_name);
+                let is_input_object = matches!(object_type.kind, ObjectKind::InputObject);
+
+                maybe_docs(out, object_type.docs, "")?;
+                writeln!(out, "export type {ts_type_name} = {{")?;
+
+                if let ObjectKind::Object = object_type.kind {
+                    writeln!(out, "{INDENT}__typename?: '{graphql_type_name}';")?;
```

Comment:
By the looks of things you might not need it I've got an indenting `Write` impl here you could steal: https://github.com/obmarg/cynic/blob/main/cynic/src/queries/indent.rs#L3-L39

Reply:
So `indented(format!("__typename?: '{graphql_type_name}')`? I don't have a strong opinion. In this file there's only ever one level of indentation, so I felt fine without a Display wrapper, but whatever is most readable is good.

Reply:
(or in some cases:

```
let out = indented(out);
writeln!(out, "whatever");
writeln!(out, "whatever");
```

But anyway, feel free to ignore this comment, may not be worthwhile

Reply:
(discussed on slack: tried it, adopted it, it's a nice win in terms of code organization and size, thanks!)